policy/daemon.go

```diff
@@ -317,11 +316,28 @@ func (p *policyd) fetchPolicy(ctx context.Context, domain string) (*SignedPolicy
 		glg.Warn(errors.Wrap(err, "error body.close"))
 	}
 
-	// set eTag cache
-	eTag := res.Header.Get("ETag")
-	if eTag != "" {
-		glg.Debugf("Setting ETag %v for domain %s", eTag, domain)
-		p.etagCache.SetWithExpire(domain, &etagCache{eTag, sp}, p.etagExpTime)
+	// verify policy expiry
+	var etagValidDur time.Duration
+	if sp.SignedPolicyData.Expires == nil {
+		glg.Warnf("Warning: policy expiry not set, domain: %v", domain)
```

Comment:
if expires is nil, maybe it is better to return error directly instead?

Reply:
no comments...

Reply:
I think for safety reason it is better to return error directly...

Reply:
remove etagExpTime, return error